gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/ddl/catalog/GfxdSystemProcedures.java

```diff
@@ -2014,6 +2014,33 @@ public static void CANCEL_STATEMENT(String statementUUID)
     }
   }
 
+  public static int CHECK_TABLE_EX(String schema, String table) throws
+      SQLException {
+    Object[] params = null;
+
+    // just add one data store member id as 3rd param on which
+    // we will verify global index region size with base table size
+    if (GemFireXDUtils.getMyVMKind().isStore()) {
+      params = new Object[]{schema, table, Misc.getMyId()};
+    } else {
+      Set<DistributedMember> dataStores = GfxdMessage.getAllDataStores();
+      DistributedMember m = dataStores.iterator().next();
+      params = new Object[]{schema, table, m};
+    }
+
+    try {
+      if (GemFireXDUtils.getMyVMKind().isStore()) {
+        GfxdSystemProcedureMessage.SysProcMethod.checkTableEx.processMessage(params,
+            Misc.getMyId());
```

Comment:
We should invoke check in parallel on all the nodes including this store vm. Since it is a time taking op so it is all the more important. Probably just including this vm id in the send target inside publishMessage.